src/adc_dma.cpp

```diff
@@ -185,6 +230,53 @@ void update_measurements()
     // else: keep previous setting
 }
 
+void lv_voltage_alert()
+{
+    // disable any sort of input
+#if FEATURE_DCDC_CONVERTER
+    dcdc.emergency_stop();
+#endif
+#if FEATURE_PWM_SWITCH
+    pwm_switch.emergency_stop();
+#endif
+    // do not use enter_state function, as we don't want to wait entire recharge delay
+    charger.state = CHG_STATE_IDLE;
+
+    // disable load output
+    if (dev_stat.has_error(ERR_BAT_OVERVOLTAGE)) {
+        load.emergency_stop(LOAD_STATE_OFF_OVERVOLTAGE);
+    }
+    else if (dev_stat.has_error(ERR_BAT_UNDERVOLTAGE)) {
+        load.emergency_stop(LOAD_STATE_OFF_SHORT_CIRCUIT);
+    }
+    else {
+        load.emergency_stop(LOAD_STATE_DISABLED);
+    }
+    charger.port->neg_current_limit = 0;
+}
+
+void adc_set_alerts()
+{
+    int vcc = VREFINT_VALUE * VREFINT_CAL /
+        (adc_filtered[ADC_POS_VREF_MCU] >> (4 + ADC_FILTER_CONST));
+
+    // LV side (battery) overvoltage alert
+    adc_alerts[ADC_POS_V_BAT].upper_limit =
```

Comment:
Does the upper_limit and the lower_limit change at run time? Currently, this is updated every 1 second. I thought this need to be set only once for a configuration.